mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -107,37 +111,55 @@ public void backupAccount(User user, OutputStream destination) throws IOExceptio
-        restoreEntries(source, session, oldToNewMailbox);
+        restoreEntries(source, session);
     }
 
-    private void restoreEntries(InputStream source, MailboxSession session, Map<SerializedMailboxId, MessageManager> oldToNewMailbox) throws IOException {
+    private void restoreEntries(InputStream source, MailboxSession session) throws IOException {
         try (MailArchiveIterator archiveIterator = archiveLoader.load(source)) {
-            archiveIterator.forEachRemaining(Throwing.consumer(entry -> {
-                restoreEntryWithErrorHandling(session, oldToNewMailbox, entry);
-            }));
-        }
-    }
+            ImmutablePair<List<MailboxWithAnnotationsArchiveEntry>, Optional<MessageArchiveEntry>> mailboxesAndFirstMessage = readMailboxes(archiveIterator);
 
-    private void restoreEntryWithErrorHandling(MailboxSession session, Map<SerializedMailboxId, MessageManager> oldToNewMailbox, MailArchiveEntry entry) throws MailboxException {
-        try {
-            restoreEntry(session, oldToNewMailbox, entry);
-        } catch (IOException e) {
-            LOGGER.error("Error when restoring message, continue ", e);
+            Map<SerializedMailboxId, MessageManager> oldToNewMailbox = restoreMailboxes(session, mailboxesAndFirstMessage);
+            Optional<MessageArchiveEntry> firstMessage = mailboxesAndFirstMessage.right;
+
+            firstMessage.ifPresent(message -> restoreNonMailboxEntry(session, oldToNewMailbox, message));
+            archiveIterator.forEachRemaining(entry -> restoreNonMailboxEntry(session, oldToNewMailbox, entry));
         }
     }
 
-    private void restoreEntry(MailboxSession session, Map<SerializedMailboxId, MessageManager> oldToNewMailbox, MailArchiveEntry entry) throws MailboxException, IOException {
+    private void restoreNonMailboxEntry(MailboxSession session, Map<SerializedMailboxId, MessageManager> oldToNewMailbox, MailArchiveEntry entry) {
         switch (entry.getType()) {
+            case MESSAGE:
+                MessageArchiveEntry messageArchiveEntry = (MessageArchiveEntry) entry;
+                restoreMessageEntryWithErrorHandling(session, oldToNewMailbox, messageArchiveEntry);
+                break;
             case MAILBOX:
-                restoreMailboxEntry(session, oldToNewMailbox, (MailboxWithAnnotationsArchiveEntry) entry);
+                MailboxWithAnnotationsArchiveEntry mailboxArchiveEntry = ((MailboxWithAnnotationsArchiveEntry) entry);
+                LOGGER.error(getMailboxEntryInWrongPositionErrorMessage(mailboxArchiveEntry));
                 break;
-            case MESSAGE:
-                restoreMessageEntry(session, oldToNewMailbox, (MessageArchiveEntry) entry);
+            case UNKNOWN:
+                String entryName = ((UnknownArchiveEntry) entry).getEntryName();
+                LOGGER.error("unknow entry found in zip :" + entryName);
                 break;
         }
     }
 
+    @NotNull
```

Comment:
We do not put it usually

Reply:
Removed, intellij likes to put that when extracting methods...

Reply:
I have the same problem, after a quick search: https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html